commercelayer/resource_payment_method.go

```diff
@@ -0,0 +1,257 @@
+		ReadContext:   resourcePaymentMethodReadFunc,
+		CreateContext: resourcePaymentMethodCreateFunc,
+		UpdateContext: resourcePaymentMethodUpdateFunc,
+		DeleteContext: resourcePaymentMethodDeleteFunc,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The payment method unique identifier",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"type": {
+				Description: "The resource type",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"attributes": {
+				Description: "Resource attributes",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Required:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"payment_source_type": {
+							Description: "The payment source type, can be one of: 'AdyenPayment', 'BraintreePayment', " +
+								"'CheckoutComPayment', 'CreditCard', 'ExternalPayment', 'KlarnaPayment', 'PaypalPayment', " +
+								"'StripePayment', or 'WireTransfer'.",
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"currency_code": {
+							Description: "The international 3-letter currency code as defined by the ISO 4217 standard. " +
+								"Required, unless inherited by market",
+							Type:     schema.TypeString,
+							Required: true,
+						},
+						"moto": {
+							Description: "Send this attribute if you want to mark the payment as MOTO, " +
+								"must be supported by payment gateway.",
+							Type:     schema.TypeBool,
+							Optional: true,
+							Default:  false,
+						},
+						"price_amount_cents": {
+							Description: "The payment method's price, in cents.",
+							Type:        schema.TypeInt,
+							Required:    true,
+						},
+						"reference": {
+							Description: "A string that you can use to add any external identifier to the resource. This " +
+								"can be useful for integrating the resource to an external system, like an ERP, a " +
+								"marketing tool, a CRM, or whatever.",
+							Type:     schema.TypeString,
+							Optional: true,
+						},
+						"reference_origin": {
+							Description: "Any identifier of the third party system that defines the reference code",
+							Type:        schema.TypeString,
+							Optional:    true,
+						},
+						"metadata": {
+							Description: "Set of key-value pairs that you can attach to the resource. This can be useful " +
+								"for storing additional information about the resource in a structured format",
+							Type: schema.TypeMap,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional: true,
+						},
+					},
+				},
+			},
+			"relationships": {
+				Description: "Resource relationships",
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				MinItems:    1,
+				Optional:    true,
```

Comment:
Relationships are required here, as we need `payment_gateway_id` to be set. Otherwise you could still leave out relationships and have a valid terraform